client-ts/Microsoft.AspNetCore.SignalR.Client.TS.Tests/HubConnection.spec.ts

```diff
@@ -401,10 +401,15 @@ describe("HubConnection", () => {
 
             // Typically this would be called by the transport
             // triggers observer.error()
-            connection.onclose(new Error("Connection lost"));
+            try {
+                connection.onclose(new Error("Connection lost"));
+            } catch (err) {
+                return Promise.reject(err);
+            }
+            return Promise.resolve();
         });
 
-        it("does not require complete function registered", async () => {
+        it("does not require complete function registered", (): Promise<any> => {
```

Comment:
I would leave this test as it was before - because what you have is semantically identical so there is no need to make it more complicated. Just add a comment saying that the expectation is that connection.receive does not throw.
(I focused only on the first test and was not clear that this one was fine - my comment about `async` was just an observation).